src/osgUtil/TriStripVisitor.cpp

```diff
@@ -607,7 +607,7 @@ void TriStripVisitor::mergeTriangleStrips(osg::Geometry::PrimitiveSetList& primi
             osg::PrimitiveSet* ps = primitives[i].get();
             // remove null primitive sets and all primitives that have been merged
             // (i.e. all TRIANGLE_STRIP DrawElements)
-            if (!ps || (ps && ps->getMode() == osg::PrimitiveSet::TRIANGLE_STRIP))
+            if (!ps || (ps->getMode() == osg::PrimitiveSet::TRIANGLE_STRIP))
```

Comment:
This change shows a mis-reading of the original code, it's a subtle bug that's introduced.  Keeping reading it until you spot why the original code is correct and the change introduces a potential crash.

Reply:
umm not sure why, still I don't see it, but ok. Nevermind on this.

Reply:
Second reading, now thing you are right.  The code is an oddity because it's catching the possibility of ps being null and if it's not null then only passing if the Mode is TRIANGLE_STRIP.  The normal pattern is if (ptr && some_test_on(ptr)), so it's really easy to mis-read a block that is if (!ptr || some_test_on_ptr). 

The original code uses the explicit (ptr && some_test_on(ptr) pattern which is sound by itself. The additional !ptr check confuses the code.  The revised code is correct but requires one to take extra care over as it doesn't fit the usual pattern.

The fact that we are even discussing this shows that original code and the revised code are both easy to confuse, I really don't like either case. Possible one could just make the assumption that all primtives[i].get() entries will be non null, so the whole catch of null values could be redundant.

Reply:
Agree, removing !ps is a better fix.

Reply:
I have just checked in the removal of !ps ||, so the code now reads like the if usage above and will be safe.

Reply:
@robertosfield I think we introduced a bug here. because if !ps the code should enter to if-block also. The way is fixed introduces a bug, as it will never enter the block.

Reply:
The new code is correct, far more readable.

If there a null primitives sets in the data being processed the new implementation handles this fine, it just doesn't remove them from the list.  The new code is far easier to read.  Case closed - I have far more important things to go on with.  The issue is closed.